tests/mmptest/src/CodeStrippingTests.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using NUnit.Framework;
+
+namespace Xamarin.MMP.Tests
+{
+	public class CodeStrippingTests
+	{
+		Func<string, bool> didStrip = output => output.SplitLines ().Any (x => x.Contains ("lipo") && x.Contains ("-remove"));
+
+		static TI.UnifiedTestConfig CreateStripTestConfig (string nativeStrip, string tmpDir, string additionalMMPArgs = "")
+		{
+			TI.UnifiedTestConfig test = new TI.UnifiedTestConfig (tmpDir) { };
+
+			if (!string.IsNullOrEmpty (nativeStrip))
+				test.CSProjConfig = $"<MonoBundlingExtraArgs>--native-strip={nativeStrip}{additionalMMPArgs}</MonoBundlingExtraArgs>";
+			return test;
+		}
+
+		[TestCase ("", false, true)]
+		[TestCase ("default", false, true)]
+		[TestCase ("strip", true, true)]
+		[TestCase ("skip", false, false)]
+		public void ShouldStripMonoPosixHelper (string nativeStrip, bool debugStrips, bool releaseStrips)
+		{
+			MMPTests.RunMMPTest (tmpDir =>
+			{
+				 TI.UnifiedTestConfig test = CreateStripTestConfig (nativeStrip, tmpDir);
+
+				 string buildOutput = TI.TestUnifiedExecutable (test).BuildOutput;
+				 Assert.AreEqual (debugStrips, didStrip (buildOutput), "Debug lipo usage did not match expectations");
+
+				 test.Release = true;
+				 buildOutput = TI.TestUnifiedExecutable (test).BuildOutput;
+				 Assert.AreEqual (releaseStrips, didStrip (buildOutput), "Release lipo usage did not match expectations");
```

Comment:
Maybe also test that the binary only has the desired architectures? We already have code to get the architectures of a binary file: https://github.com/xamarin/xamarin-macios/blob/bec116c273ab119c648522a28c3f650881ed4d9e/tests/mtouch/MTouch.cs#L3996